packages/api-client/src/api/serializers/product.ts

```diff
@@ -234,11 +234,15 @@ const addHostToImage = (image, config: ApiConfig) => ({
   ...image,
   attributes: {
     ...image.attributes,
-    styles: image.attributes?.styles ? image.attributes.styles.map((style) => ({
-      width: style.width,
-      height: style.height,
-      url: `${style.url.includes('http') ? '' : (config.assetsUrl || config.backendUrl)}${style.url}`
-    })) : []
+    styles: image.attributes?.styles ? image.attributes.styles.map((style) => {
+      const host = style.url.includes('http') ? '' : (config.assetsUrl || config.backendUrl);
```

Comment:
Actually one more thing - I think we should test `style.url.startsWith('http://') || style.url.startsWith('https://')`
There's a small chance that the asset link generated by rails will have a `http` somewhere in the path,  without it being the protocol.